js/localization/ldml/date.parser.js

```diff
@@ -218,6 +219,22 @@ export const getRegExpInfo = function(format, dateParts) {
 
     addPreviousStub();
 
+    patterns.every((pattern, index) => {
+        const char = pattern[0];
+        const prevPatternChar = index ? patterns[index - 1][0] : null;
+        const nextPatternChar = index < patterns.length - 1 ? patterns[index + 1][0] : null;
+        const isSingle = pattern.length === 1;
+        const isFirst = index === 0;
+        const isRegexpPart = (char) => PATTERN_REGEXPS[char] && char !== ':';
+
+        if(isSingle && isRegexpPart(char) && (isRegexpPart(prevPatternChar) || (isFirst && isRegexpPart(nextPatternChar)))) {
+            logger.warn(`Date-time formats without separators may produce unexpected results. Please add separators to the following format: ${format}.`);
```

Comment:
I suppose the warning text that Alexey Serov suggested describes the possible solutions more completely
>The ${invalidPattern} may lead to undefined parsing behavior. Consider adding separators, or replace minimum digits symbols with the fixed digits notation